nbs/methods.ipynb

```diff
@@ -44,6 +44,7 @@
     "\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #17.            res = {'mean': np.matmul(S @ P, y_hat)}
</pre><p><code>np.matmul(S @ P, y_hat)</code>  is equivalent to <code>S.dot(P.dot(y_hat))</code>  when using numpy arrays? If that's the case, we could use <code>{'mean': S.dot(P.dot(y_hat))}</code> for all cases and leave the transformation to sparse arrays to the methods (to avoid adding a new argument).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/hierarchicalforecast/pull/109/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
Sounds good, I added an [issue](https://github.com/Nixtla/hierarchicalforecast/issues/110) to change the corresponding matrix multiplication for the dot method.

I would keep the tests and the sparsity parameter momentarily to track these and other method's speed gains. Currently is a default parameter that is not shown to the higher-level methods.